### DISCUSSION TITLE
Trying to understand `by` from the examples

### DISCUSSION THREAD

**Discussion by avillega** [hylo-lang/hylo#1427] [Q&A] [answered]
Hi! thanks a lot for the effort you are putting in this language, it looks fantastic!
I was going through the language tour and kept seeing a patter where the name of the argument of a functions is preceded by a `_` or by `by` I couldn't find what does this "labels" mean. By how it is used, it seems to be a way of referring to that argument from the caller, but I am not sure. 
a full example, from the language tour looks like:
```
fun scale(_ v: Vector2, by factor: Vector2) -> Vector2 {
  (x: v.x * factor.x, y: v.y * factor.y)
}
```
Note the `by factor: ...` and the `_ v: ...`

**Comment by kyouko-taiga** [comment, marked as answer]
Those are argument labels and they are part of a function's (or subscript's) API. When you declare a function, for each parameter you get to specify an argument label, which must be used at call site, and a parameter name, which is used in the body of the function. Those can be distinct:

```
fun foo(label name: Int) {
  // here we must use `name`
}

fun bar() {
  foo(label: 1) // here we must use `label`
}
```

You can use `_` to specify that there is no label at call site, as in the example you presented. Finally, as a shorthand, you can omit the label in the declaration, in which case it gets the same value as the parameter name. In other words, `fun f(x: Int)` is short for `fun(x x: Int)`.

All of these rules have been shamelessly copied from Swift.
